lib/hllcounter.cc

```diff
@@ -269,36 +181,37 @@ HLLCounter::HLLCounter(double error_rate, WordLength ksize)
         throw InvalidValue("Please set error rate to a value "
                            "greater than zero");
     }
-    int p = ceil(log2(pow(1.04 / error_rate, 2)));
-    this->init(p, ksize);
+    int new_p = ceil(log2(pow(1.04 / error_rate, 2)));
+    init(1 << new_p, ksize);
 }
 
-HLLCounter::HLLCounter(int p, WordLength ksize)
+HLLCounter::HLLCounter(int nc, WordLength ksize)
 {
-    this->init(p, ksize);
+    init(nc, ksize);
 }
 
-void HLLCounter::init(int p, WordLength ksize)
+void HLLCounter::init(int nc, WordLength ksize)
 {
-    this->alpha = calc_alpha(p);
-    this->p = p;
-    this->_ksize = ksize;
-    this->m = 1 << p;
-    std::vector<int> M(this->m, 0.0);
-    this->M = M;
-
-    init_raw_estimate_data();
-    init_bias_data();
+    alpha = calc_alpha(nc);
+    ncounters_log2 = floor(log2(nc));
+    ncounters = 1 << ncounters_log2;
+    _ksize = ksize;
+    std::vector<uint8_t> newc(ncounters, 0);
+    counters = newc;
```

Comment:
Why this indirection of first making a new vector and then assigning it? The goal is to have a vector of the correct size filled only with zeros?